src/OpenLoco/src/Windows/Construction/ConstructionTab.cpp

```diff
@@ -621,14 +621,10 @@ namespace OpenLoco::Ui::Windows::Construction::Construction
     // 0x0049DAF3
     static void disableTrackSlopes(Window* self, TrackObject trackObj, uint64_t disabledWidgets)
     {
-        auto maskedTrackPieces = trackObj.trackPieces & ((1 << 5) | (1 << 8));
-
-        if (maskedTrackPieces != ((1 << 5) | (1 << 8)))
+        if (!trackObj.hasFlags(TrackPieceFlags::slope | TrackPieceFlags::slopedCurve))
```

Comment:
This logic isn't quite the same. It should be two has flags calls && together. It's saying if both not (slope and curved slope). You've written if not ( slope or curved slope )